cli/pcluster/utils.py

```diff
@@ -771,6 +771,17 @@ def get_info_for_amis(ami_ids):
         error(e.response.get("Error").get("Message"))
 
 
+def validate_pcluster_version_based_on_ami_name(ami_name):
+    match = re.match(r"(.*)aws-parallelcluster-(\d+\.\d+\.\d+)(.*)", ami_name)
+    if match:
+        if ami_name.find("aws-parallelcluster-{0}".format(get_installed_version())) < 0:
```

Comment:
If the version string is followed by digits, only current approach works properly.
For example the AMI name is ami-aws-parallelcluster-2.8.1123.

Reply:
It's good that you thought about this, but I think this is not a case we need to consider. Moreover, along the same lines, the current implementation would actually cause an issue in case of using `aws-parallelcluster-2.8.1` AMI with `aws-parallelcluster-2.8.11`, if we actually had an 2.8.11 release.

I would like us to err on the side of caution and only do comparison on the extracted version.